extended/src/main/java/apoc/vectordb/VectorDb.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    private static Entity handleMapping(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metadata, List<Double> embedding) {
+        if (mapping.getProp() == null) {
+            return null;
+        }
+        if (MapUtils.isEmpty(metadata)) {
+            throw new RuntimeException("To use mapping config, the metadata should not be empty. Make sure you execute `YIELD metadata` on the procedure");
+        }
+        Map<String, Object> metaProps = new HashMap<>(metadata);
+        if (mapping.getLabel() != null) {
+            return handleMappingNode(tx, db, mapping, metaProps, embedding);
+        } else if (mapping.getType() != null) {
+            return handleMappingRel(tx, db, mapping, metaProps, embedding);
+        } else {
+            throw new RuntimeException("Mapping conf has to contain either label or type key");
+        }
+    }
+
+    private static Entity handleMappingNode(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metaProps, List<Double> embedding) {
+        String query = "CREATE CONSTRAINT IF NOT EXISTS FOR (n:%s) REQUIRE n.%s IS UNIQUE"
+                .formatted(mapping.getLabel(), mapping.getProp());
+        db.executeTransactionally(query);
+
+        try {
+            Node node;
+            try (Transaction transaction = db.beginTx()) {
+                Object propValue = metaProps.remove(mapping.getId());
+                node = transaction.findNode(Label.label(mapping.getLabel()), mapping.getProp(), propValue);
+                if (node == null && mapping.isCreate()) {
+                    node = transaction.createNode(Label.label(mapping.getLabel()));
+                    node.setProperty(mapping.getProp(), propValue);
+                }
+                if (node != null) {
+                    setProperties(node, metaProps);
+                }
+                transaction.commit();
+            }
+
+            String setVectorQuery = "CALL db.create.setNodeVectorProperty($entity, $key, $vector)";
+            setVectorProp(tx, db, mapping, embedding, node, setVectorQuery);
+            return node;
+        } catch (MultipleFoundException e) {
+            throw new RuntimeException("Multiple nodes found");
+        }
+    }
+
+    private static Entity handleMappingRel(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metaProps, List<Double> embedding) {
+        try {
+            String query = "CREATE CONSTRAINT IF NOT EXISTS FOR ()-[r:%s]-() REQUIRE (r.%s) IS UNIQUE"
+                    .formatted(mapping.getType(), mapping.getProp());
+            db.executeTransactionally(query);
+
+            // in this case we cannot auto-create the rel, since we should have to define start and end node as well
+            Relationship rel;
+            try (Transaction transaction = db.beginTx()) {
+                Object propValue = metaProps.remove(mapping.getId());
```

Comment:
should we really remove the mapping-id ? if we later return the metadata that's missing?